pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -768,7 +833,7 @@ func classifyClusterDecisions(decisions []fleetv1beta1.ClusterDecision) (selecte
 	return selected, unselected
 }
 
-func buildResourcePlacementStatus(crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot) []fleetv1beta1.ResourcePlacementStatus {
+func (r *Reconciler) buildResourcePlacementStatusAndAppliedCondition(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot, latestResourceSnapshot *fleetv1beta1.ClusterResourceSnapshot) ([]fleetv1beta1.ResourcePlacementStatus, metav1.Condition, error) {
 	res := make([]fleetv1beta1.ResourcePlacementStatus, 0, len(latestSchedulingPolicySnapshot.Status.ClusterDecisions))
 	decisions := latestSchedulingPolicySnapshot.Status.ClusterDecisions
 	selected, unselected := classifyClusterDecisions(decisions)
```

Comment:
This thread is about the section below (new line 841) and its related logic:

Reply:
For PickAll type, the scheduled condition (on policy snapshot) would not be false, as it is always fully scheduled; for the PickN type, the condition being false implies not being able to pick enough number of clusters. And in this case, we cannot pinpoint exactly which unselected clusters are the cause of the problem, so they have to be all included (or at least, included as much as possible) for the user to make an analysis.

Reply:
```
         // PlacementStatuses contains a list of placement status on the clusters that are selected by PlacementPolicy.
	// Each selected cluster according to the latest resource placement is guaranteed to have a corresponding placementStatuses.
	// In the pickN case, there are N placement statuses where N = NumberOfClusters. Some of them may not have assigned
	// clusters when we cannot fill the required number of clusters.
	// +optional
	PlacementStatuses []ResourcePlacementStatus 
```

The implementation is following the placementStatuses definition. The expected size of placementStatuses is N = NumberOfClusters (will be 5 if using the example in the comment). The other two (scheduled 3) will be used to explain why. 

have a TODO in line 893. We will populate these two by summarizing the failure reasons.
For now, we just populate the sample failures.